### ISSUE TITLE
Display a TOC for the current page and the pages beneath it on a page

### ISSUE THREAD

**Issue by khuizer** [WikiTransformationProject/wikitraccs-wikipakk-releases#13]
A much used feature within our Confluence environment is that people can put a TOC of the current page and its child pages on a page. 

I guess this is something that would probably need a separate webpart (?)

**Comment by heinrich-ulbricht** [comment]
Note: a child pages web part is coming in the next release v2.9.0

**Comment by bhouleinf** [comment]
This is awesome! Any plan for the TOC?

**Comment by heinrich-ulbricht** [comment]
@bhouleinf A TOC webpart is not on the [roadmap](https://github.com/orgs/WikiTransformationProject/projects/4) for 2025, but definitely a candidate for 2026!

**Comment by synhoub** [comment]
Any news on this? :)

**Comment by heinrich-ulbricht** [comment]
@bhouleinf The main technical blocker for this so far is progressive loading of SharePoint pages. If you open a SharePoint page with many web parts and headings, not all web parts are loaded, and not all headings will be present. Parsing the _rendered_ page would miss headings, until you scroll and they pop up. At least this is my understanding.

I know there are workarounds, but none of them are pretty, in my view.

If you know of any solution, please send a link along my way so I have proof this can work.

**Comment by synhoub** [comment]
Ah makes sense. Can you get the page content via API and then reconstruct the anchors links from there?

https://you-domain.comm/sites/your-site/_api/web/GetFileByServerRelativeUrl('/sites/your-site/SitePages/your-page.aspx')/ListItemAllFields?$select=CanvasContent1

There are no ids for your anchors in there so you would need to reconstruct them with a pattern :|

**Comment by heinrich-ulbricht** [comment]
@bhouleinf Yes, this is one of the approaches that don't feel quite right. It loads the whole page a second time (can be hundreds of KB, up to MB) - can be OK, or not. Then parsing could end up different than what the browser is rendering.

Plus: I'm not even sure if the jump to the target anchor works, if it hasn't been loaded yet, although I think I remember that this did work via some magic SharePoint applies.

Would have to dive into this. I know it can be solved. Just have to find the best of the not-so-ideal approaches. But this is the reason why I'm a bit hesitant. I guess I'm still hoping MS is giving us something ^^

**Comment by heinrich-ulbricht** [comment]
I'm looking into this (TOC web part for current page) as this would be a great addition.

**Comment by heinrich-ulbricht** [comment]
The table of contents web part is entering testing phase. I expect a new WikiPakk release to be out within the next 2 weeks.

**Comment by synhoub** [comment]
Oh nice! You found a way that is not too hacky?

**Comment by heinrich-ulbricht** [comment]
@synhoub Actually, yes! It's live now in release 2.16 and available on the [marketplace](https://marketplace.microsoft.com/en-us/product/office/wa200005439?tab=overview)!

See some more information and screenshots in this discussion: https://github.com/WikiTransformationProject/wikitraccs-releases/discussions/182

I have yet to update the docs and record a demo video.

**Comment by synhoub** [comment]
Goood thank you! We will deploy it soon in our tenant.
